pkg/controller/servicediscovery/servicediscovery_controller.go

```diff
@@ -90,35 +94,31 @@ func (r *ReconcileServiceDiscovery) Reconcile(request reconcile.Request) (reconc
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			daemonSet := &appsv1.DaemonSet{}
+			pod := &corev1.Pod{}
 			opts := []client.DeleteAllOfOption{
 				client.InNamespace(request.NamespacedName.Namespace),
 				client.MatchingLabels{"app": appName},
 			}
-			err := r.client.DeleteAllOf(context.TODO(), daemonSet, opts...)
+			err := r.client.DeleteAllOf(context.TODO(), pod, opts...)
 			return reconcile.Result{}, err
 		}
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
 	lightHouseAgent := newLighthouseAgent(instance)
-	if _, err = helpers.ReconcileDaemonSet(instance, lightHouseAgent, reqLogger,
+	if _, err = helpers.ReconcilePod(instance, lightHouseAgent, reqLogger,
```

Comment:
This should be a `Deployment` not a `Pod`.

Reply:
Done.